File: PYTHON/film.py
from media import Media

class Film(Media):
    def __init__(self, media_id=0, judul="", tahun=0, hak_cipta="", genre="", durasi_menit=0, rumah_produksi=""):
        super().__init__(media_id, judul, tahun, hak_cipta)
        self._genre = genre
        self.set_durasi_menit(durasi_menit)
        self._rumah_produksi = rumah_produksi

    def get_genre(self): return self._genre
    def set_genre(self, genre): self._genre = genre

    def get_durasi_menit(self): return self._durasi_menit
    def set_durasi_menit(self, durasi_menit): self._durasi_menit = max(durasi_menit, 0)

    def get_rumah_produksi(self): return self._rumah_produksi
    def set_rumah_produksi(self, rumah_produksi): self._rumah_produksi = rumah_produksi

File: JAVA/Film.java
package JAVA;

public class Film extends Media {
    private String genre;
    private int durasiMenit;
    private String rumahProduksi;
    /* Constractor */
    public Film() {}

    public Film(int id, String judul, int tahun, String hakCipta, String genre, int durasiMenit, String rumahProduksi) {
        super(id, judul, tahun, hakCipta);
        this.genre = genre;
        this.setDurasiMenit(durasiMenit);
        this.rumahProduksi = rumahProduksi;
    }
    /* Method */
    public String getGenre() { return genre; }
    public void setGenre(String genre) { this.genre = genre; }

    public int getDurasiMenit() { return durasiMenit; }
    public void setDurasiMenit(int durasiMenit) {
        this.durasiMenit = Math.max(durasiMenit, 0);
    }

    public String getRumahProduksi() { return rumahProduksi; }
    public void setRumahProduksi(String rumahProduksi) { this.rumahProduksi = rumahProduksi; }
}